npc/re/other/LapineEnchants.txt

```diff
@@ -0,0 +1,2071 @@
+		.@value = rand(5,30);
+	} else if (.@r < 4600) { // To Fire Element Damage +3~20%
+		.@option = RDMOPT_DAMAGE_PROPERTY_FIRE_TARGET;
+		.@value = rand(5,30);
+	} else if (.@r < 5500) { // To Wind Element Damage +3~20%
+		.@option = RDMOPT_DAMAGE_PROPERTY_WIND_TARGET;
+		.@value = rand(5,30);
+	} else if (.@r < 6400) { // To Poison Element Damage +3~20%
+		.@option = RDMOPT_DAMAGE_PROPERTY_POISON_TARGET;
+		.@value = rand(5,30);
+	} else if (.@r < 7300) { // To Holy Element Damage +3~20%
+		.@option = RDMOPT_DAMAGE_PROPERTY_SAINT_TARGET;
+		.@value = rand(5,30);
+	} else if (.@r < 8200) { // To Shadow Element Damage +3~20%
+		.@option = RDMOPT_DAMAGE_PROPERTY_DARKNESS_TARGET;
+		.@value = rand(5,30);
+	} else if (.@r < 9100) { // To Ghost Element Damage +3~20%
+		.@option = RDMOPT_DAMAGE_PROPERTY_TELEKINESIS_TARGET;
+		.@value = rand(5,30);
+	} else { // To Undead Element Damage +3~20%
+		.@option = RDMOPT_DAMAGE_PROPERTY_UNDEAD_TARGET;
+		.@value = rand(5,30);
+	}
+
+	set getarg(0), .@option;
+	set getarg(1), .@value;
+
+	// Option:2
+	.@r = rand(10000);
+	if (.@r < 100) { // ATK +1~5%
+		.@option = RDMOPT_VAR_ATKPERCENT;
+		.@value = rand(1,5);
+	} else if (.@r < 700) { // Critical Damage +3~10%
+		.@option = RDMOPT_DAMAGE_CRI_TARGET;
+		.@value = rand(3,10);
+	} else if (.@r < 1600) { // To Formless Damage +3~20%
+		.@option = RDMOPT_RACE_DAMAGE_NOTHING;
+		.@value = rand(3,20);
+	} else if (.@r < 2500) { // To Undead Race damage +3~20%
+		.@option = RDMOPT_RACE_DAMAGE_UNDEAD;
+		.@value = rand(3,20);
+	} else if (.@r < 3400) { // To Beast Race Damage +3~20%
+		.@option = RDMOPT_RACE_DAMAGE_ANIMAL;
+		.@value = rand(3,20);
+	} else if (.@r < 4300) { // To Plant Race Damage +3~20%
+		.@option = RDMOPT_RACE_DAMAGE_PLANT;
+		.@value = rand(5,30);
+	} else if (.@r < 5200) { // To Insect Race Damage +3~20%
+		.@option = RDMOPT_RACE_DAMAGE_INSECT;
+		.@value = rand(5,30);
+	} else if (.@r < 6100) { // To Fish Race Damage +3~20%
+		.@option = RDMOPT_RACE_DAMAGE_FISHS;
+		.@value = rand(5,30);
+	} else if (.@r < 7000) { // To Demon Race Damage +3~20%
+		.@option = RDMOPT_RACE_DAMAGE_DEVIL;
+		.@value = rand(5,30);
+	} else if (.@r < 7900) { // To Demi-Human Race Damage +3~20%
+		.@option = RDMOPT_RACE_DAMAGE_HUMAN;
+		.@value = rand(5,30);
+	} else if (.@r < 8800) { // To Angel Race Damage +3~20%
+		.@option = RDMOPT_RACE_DAMAGE_ANGEL;
+		.@value = rand(5,30);
+	} else if (.@r < 9700) { // To Dragon Race Damage +3~20%
+		.@option = RDMOPT_RACE_DAMAGE_DRAGON;
+		.@value = rand(5,30);
+	} else if (.@r < 9800) { // Increase damage to normal class 1~10%
+		.@option = RDMOPT_CLASS_DAMAGE_NORMAL_TARGET;
+		.@value = rand(1,10);
+	} else if (.@r < 9900) { // Increase damage to boss class 1~10%
+		.@option = RDMOPT_CLASS_DAMAGE_BOSS_TARGET;
+		.@value = rand(1,10);
+	} else { // Indestructible
+		.@option = RDMOPT_WEAPON_INDESTRUCTIBLE;
+		.@value = 1;
+	}
+
+	set getarg(2), .@option;
+	set getarg(3), .@value;
```

Comment:
The call in `item_upgrade.yml` could be simplified eventually

```
callfunc("F_Lapine_Ancient_Hero_Bravery", .@opts, .@vals);
```
The function:
```
set getelementofarray(getarg(0), 0), .@option; // .@opts index 0
set getelementofarray(getarg(1), 0), .@value;

set getelementofarray(getarg(0), 1), .@option; // .@opts index 1
set getelementofarray(getarg(1), 1), .@value;
```

Reply:
I dislike to call that reduntant functions. it looks neat & fancy, but not that the reason.